### DISCUSSION TITLE
Small problem with SpaceEngineersFinder task

### DISCUSSION THREAD

**Discussion by overdeveloped** [malforge/mdk2#26] [General]
Hi. I have been getting a great deal of value from your tool. It is a godsend. However, after a few months break I've gone back in and found that it's stopped working. When creating a new solution using the MDK2 template in VS I imediately get an error:

Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	MSB4018	The "SpaceEngineersFinder" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Mdk2.References.SpaceEngineersFinder.LoadFromIni(String localIniFileName)
   at Mdk2.References.SpaceEngineersFinder.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	AnalogThrottleBasic	C:\Users\exper\.nuget\packages\mal.mdk2.references\2.2.3\build\Mal.Mdk2.References.targets	14		

It seems to be looking for this missing dll in the nuget package and not my own code. Have you seen this before?

Kind regards.

**Comment by malware-dev** [comment]
It's _supposed_ to. It's a separate task, and that task needs the immutable assembly.

I have _no_ idea why it doesn't find it though...

**Reply by overdeveloped** [reply]
Thanks for getting back to me. The problem was solved by updating to the latest version of VS 2022

**Reply by malware-dev** [reply]
Ah! Thank you for letting me know!
